### ISSUE TITLE
Caching problem with --new-semantic-analyzer and --warn-unused-ignores

### ISSUE THREAD

**Issue by srittau** [python/mypy#7025]
With mypy 0.710 (Python 3.7.2) and the new semantic analyzer, I got a strange caching problem. Consider a directory with three files:

```python
# __init__.py (empty)

# foo.py
from typing import Any

class desc:
    def __get__(self, _: Any, __: Any = None) -> int:
        return 42

class Base:
    @property
    def foo(self) -> int:
        raise NotImplementedError()

class Sub(Base):
    foo = desc(42)  # type: ignore

# bar.py
from foo import Sub

Sub().foo
Sub().foo
```

Running mypy in this directory for the first time with `PYTHONPATH=. mypy --new-semantic-analyzer --warn-unused-ignores bar.py` results in the following (expected) output:

```
bar.py:3: error: Cannot determine type of 'foo'
bar.py:4: error: Cannot determine type of 'foo'
```

Changing `bar.py` to this:

```python
from foo import Sub

Sub().foo  # type: ignore
Sub().foo
```

and rerunning mypy with the command above now results in this error message:

```
bar.py:3: note: unused 'type: ignore' comment
```

Note that the problem in line 4 is not reported anymore.

Restoring the original `bar.py` and rerunning mypy for a third time will pass without any error message.

Deleting `.mypy_cache` will start this process from the beginning.

**Comment by ilevkivskyi** [comment]
Thanks for reporting! I think this is important to investigate.
